benchmarking/src/cloudflare.ts

```diff
@@ -0,0 +1,101 @@
+import nodeFsPromises from "node:fs/promises";
+import nodeFs from "node:fs";
+import nodePath from "node:path";
+import nodeChildProcess from "node:child_process";
+import nodeUtil from "node:util";
+
+const promiseExec = nodeUtil.promisify(nodeChildProcess.exec);
+
+await ensureWranglerSetup();
+
+/**
+ * Collects name and absolute paths of apps (in this repository) that we want to benchmark
+ *
+ * @returns Array of objects containing the app's name and absolute path
+ */
+export async function collectAppPathsToBenchmark(): Promise<
+  {
+    name: string;
+    path: string;
+  }[]
+> {
+  const allExampleNames = await nodeFsPromises.readdir("../examples");
+
+  /**
+   * Example applications that we don't want to benchmark
+   *
+   * Currently we only want to skip the `vercel-commerce` example, and that's simply
+   * because it requires a shopify specific setup and secrets.
+   */
+  const exampleAppsNotToBenchmark = new Set(["vercel-commerce"]);
+
+  const examplePaths = allExampleNames
+    .filter((exampleName) => !exampleAppsNotToBenchmark.has(exampleName))
+    .map((exampleName) => ({
+      name: exampleName,
+      path: nodePath.resolve(`../examples/${exampleName}`),
+    }));
+
+  return examplePaths;
+}
+
+/**
+ * Builds an application using their "build:worker" script
+ * (an error is thrown if the application doesn't have such a script)
+ *
+ * @param dir Path to the application to build
+ */
+export async function buildApp(dir: string): Promise<void> {
+  const packageJsonPath = `${dir}/package.json`;
+  if (!nodeFs.existsSync(packageJsonPath)) {
+    throw new Error(`Error: package.json for app at "${dir}" not found`);
+  }
+
+  const packageJsonContent = JSON.parse(await nodeFsPromises.readFile(packageJsonPath, "utf8"));
+
+  if (!("scripts" in packageJsonContent) || !("build:worker" in packageJsonContent.scripts)) {
```

Comment:
`if (!packageJsonContent.scripts?.[BUILD_SCRIPT_NAME])` and reuse the const for `command`?

Reply:
ah yeah good catch 👍